src/sync.rs

```diff
@@ -665,7 +665,9 @@ const fn buffer_index(i: usize) -> usize {
     (((usize::BITS + 1 - (i + 1).leading_zeros()) >> 1) - 1) as usize
 }
 
-const NUM_BUFFERS: usize = (usize::BITS >> 2) as usize;
+/// Each buffer covers 2 bits of address space, so we need half as many
+/// buffers as bits in a usize.
+const NUM_BUFFERS: usize = (usize::BITS / 2) as usize;
```

Comment:
Good catch. I had an off by one here in my head because I always calculated the total buffer size including the buffer at index `NUM_BUFFERS`.

Maybe we should grow buffers faster instead of increasing the number of buffers?

Reply:
The buffers are already growing at 4x each time it grows, which is quite aggresive. Not increasing the buffer size would require an 8x growth factor, which just seems excessive to me. Especially given that a few non-breaking versions ago (1.7.1/1.8.1) this was only growing 2x at each cutover point.

Reply:
Note that this also means that in some cases 8 times more storage would be used than is strictly required to store the elements. I think that that is not worth the small reduction in number of pointers used.